src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -2,12 +2,34 @@
  * @output wp-includes/js/wp-emoji-loader.js
  */
 
-( function( window, document, settings ) {
-	var src, ready, ii, tests;
+/**
+ * Emoji Settings as exported in PHP via _print_emoji_detection_script().
+ * @typedef WPEmojiSettings
+ * @type {object}
+ * @property {!object} source
+ * @property {!string} source.concatemoji
+ * @property {!string} source.twemoji
+ * @property {!string} source.wpemoji
+ */
+
+/**
+ * IIFE to detect emoji support and load Twemoji if needed.
+ *
+ * @param {Window} window
+ * @param {Document} document
+ * @param {WPEmojiSettings} settings
+ */
+( async function( window, document, settings ) {
```

Comment:
I'm getting an error here:

```
Running "jsvalidate:build" (jsvalidate) task
Validating build/wp-includes/js/wp-emoji-loader.js  
>> Line 22: Unexpected token function
```

But I'm not immediately clear why.

Reply:
It may be due to Esprima not supporting ECMAScript 2017 yet? Or maybe there's a way to enable support in the `esprimaOptions` config:

https://github.com/WordPress/wordpress-develop/blob/3e2121c83de37335bcda944a09c2d1a8f11dab7b/Gruntfile.js#L1038-L1043

I haven't been able to quickly determine what the possible options are.

Reply:
The version of [grunt-jsvalidate](https://www.npmjs.com/package/grunt-jsvalidate) used in core is 0.2.2, which is the latest version available. It was last updated _10 years ago_. What's more is that the [GitHub repo](https://github.com/ariya/grunt-jsvalidate) has been archived. So clearly we need to use something else to check for syntax errors in JS files.

Reply:
I responded to [your question about this in slack](https://wordpress.slack.com/archives/C5UNMSU4R/p1686977423731239), but it may also be time for us to consider removing this validation step entirely at this point. Seems like perhaps this old validation task is causing more friction in our build process than it is helping.

Reply:
For the sake of expediency for beta1, I'm going to try to make this ES3-compatible and then we can revisit taking advantage of ES6+.